utils/session/index.ts

```diff
@@ -47,6 +53,27 @@ export const setAgentSession = async (
   await session.save();
 };
 
+export const cleanAgentSession = async (session: IronSession) => {
+  session.user = {};
+  await session.save();
+};
+
+/**
+ * Store Siren for redirections
+ * @param session
+ */
+
+export const setSirenFrom = async (session: IronSession, sirenFrom: string) => {
+  session.sirenFrom = sirenFrom;
+  await session.save();
+};
+
+export const getSirenFrom = (session: IronSession) => session.sirenFrom;
+
+export const cleanSirenFrom = async (session: IronSession) => {
+  session.sirenFrom = undefined;
```

Comment:
```suggestion
  delete session.sirenFrom;
```
I find it more bulletproof, otherwise `sirenFrom in session` would return `true`. But it's not crucial, and if your code typecheck, it must mean that I'm missing something and that I may be mistaken…